rasterio/_base.pyx

```diff
@@ -680,7 +680,7 @@ cdef class DatasetBase:
 
         Returns
         -------
-        list of str
+        tuple
```

Comment:
@sgillies, thoughts on using this syntax?
https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html
```suggestion
        tuple[str | None, ...]
```

Reply:
Sure! The meaning of the ellipsis isn't as clear as I'd like. The intent of the code is to replace empty strings (which GDAL returns) by None. I don't think I'm necessarily right, but I have in the past preferred None to an empty string. I feel like it says "there is no description", whereas "" says "there is a description but no characters are in it".

Reply:
Makes sense to me 👍